src/snowflake/connector/errors.py

```diff
@@ -66,9 +68,9 @@ def __init__(
             if self.sqlstate != "n/a":
                 if not already_formatted_msg:
                     if logger.getEffectiveLevel() in (logging.INFO, logging.DEBUG):
-                        self.msg = f"{self.errno:06d} ({self.sqlstate}): {self.sfqid}: {self.msg}"
+                        self.msg = f"{self.errno:06d} ({self.sqlstate}): {self.sfqid}: {self.msg}: {self.query}"
```

Comment:
@sfc-gh-mkeller could I get your input on whether this could be considered behavior change? We want to add `query` into `Errors` generated by python connector (more context [here](https://github.com/snowflakedb/snowpark-python/pull/674#issuecomment-1428488743)) so that `query` is easily accessible when debugging issues and does not get lost.

Reply:
However; it does affect `__str__`

```
>>> str(e)
'000111: this is a test error'
```

Reply:
As a middle ground could we not add the query to `msg` but have it as a field of `Error`? For the short-term at least